packages/typescript-estree/src/convert.ts

```diff
@@ -1058,23 +1058,6 @@ export class Converter {
       }
 
       case SyntaxKind.PropertyAssignment: {
-        // eslint-disable-next-line deprecation/deprecation
-        const { questionToken, exclamationToken } = node;
-
-        if (questionToken) {
-          this.#throwError(
-            questionToken,
-            'A property assignment cannot have a question token.',
-          );
-        }
-
-        if (exclamationToken) {
-          this.#throwError(
-            exclamationToken,
-            'A property assignment cannot have an exclamation token.',
-          );
-        }
```

Comment:
FYI @fisker - TypeScript 5.0 removed a bunch of these deprecated properties. That's why I removed these lines. Let me know if there's a better way to handle the removal please!

Reply:
The diff is so big, I can't see what's actually removed.

As I understand, `illegalDecorators` also removed in v5, in Prettier I made [this change](https://github.com/prettier/prettier/pull/13819/files#diff-2aa2b00967877a193c98032139d7685caef7777c295e3da65d1b5bc3e1bd668fL26) to make it work.

Reply:
Funnily enough even though TS removed the properties from the types - TS doesn't actually parser error on the bad syntax - they're still semantic errors :frowning:
https://www.typescriptlang.org/play?ts=5.1.0-dev.20230313#code/C4TwDgpgBAKlC8UDeAoKUBmB7LAuKAjAoQNwoC+KAxlgHYDOwUAHvnIkpUA

So I've re-added the errors and patched TS types to add back the removed properties in 6586d98b944a5d5e37e7b469e7e044389ed919be